Python/compile.c

```diff
@@ -1002,27 +1008,33 @@ PyCompile_OpcodeStackEffect(int opcode, int oparg)
         case IMPORT_FROM:
             return 1;
 
+        /* Jumps */
         case JUMP_FORWARD:
-        case JUMP_IF_TRUE_OR_POP:  /* -1 if jump not taken */
-        case JUMP_IF_FALSE_OR_POP:  /*  "" */
         case JUMP_ABSOLUTE:
             return 0;
-
+        case JUMP_IF_FALSE_OR_POP:
+        case JUMP_IF_TRUE_OR_POP:
+            /* 0 if jumped */
+            return -1;
         case POP_JUMP_IF_FALSE:
         case POP_JUMP_IF_TRUE:
             return -1;
 
         case LOAD_GLOBAL:
             return 1;
 
-        case CONTINUE_LOOP:
+        /* Exception handling */
+        case SETUP_FINALLY:
+            /* Reserve 3 entries for the new exception
+             * + 3 others for the previous exception state */
+            return 6;
+        case BEGIN_FINALLY:
+            /* The stack has been reserved by SETUP_FINALLY */
             return 0;
-        case SETUP_LOOP:
+        case CALL_FINALLY:
             return 0;
-        case SETUP_EXCEPT:
-        case SETUP_FINALLY:
-            return 6; /* can push 3 values for the new exception
-                + 3 others for the previous exception state */
+        case POP_FINALLY:
+            return -6;
```

Comment:
Why is this statically -6? This would deserve a comment.

Reply:
Done.